.github/workflows/unit-tests.yaml

```diff
@@ -0,0 +1,84 @@
+# yamllint disable rule:line-length
+# yamllint disable rule:braces
+
+name: Unit & Integration Tests
```

Comment:
the name of the file is `unit-tests.yaml`, but here we are saying this is also for Integration tests. Let's rename the file to `tests.yaml`? Does it make sense? (I'm not sure as we still have e2e tests in `ci.yaml`)

Reply:
`tests.yaml` sounds good to me.